ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -872,19 +872,22 @@ open class NavigationBar: UINavigationBar, TokenizedControlInternal, TwoLineTitl
         updateShadow(for: navigationItem)
     }
 
-    private func updateFakeCenterTitleConstraints() {
-        titleViewConstraint?.isActive = false
+    private func updateTitleViewConstraints() {
+        if var titleViewConstraints = titleViewConstraints {
+            NSLayoutConstraint.deactivate(titleViewConstraints)
+            titleViewConstraints.removeAll()
+        }
 
-        let newTitleViewConstraint: NSLayoutConstraint
-        if !usesLeadingTitle && systemWantsCompactNavigationBar {
-            // If we're drawing our own system-style bar above the OS bar, align our title with the OS's
-            newTitleViewConstraint = titleView.centerXAnchor.constraint(equalTo: centerXAnchor)
-        } else {
-            // Otherwise, keep `self.titleView` leading-justified
-            newTitleViewConstraint = preTitleSpacerView.widthAnchor.constraint(equalToConstant: 0)
+        titleViewConstraints = [titleView.topAnchor.constraint(equalTo: topAnchor),
+                                titleView.bottomAnchor.constraint(equalTo: bottomAnchor)]
+        
+        if usesLeadingTitle {
+            titleViewConstraints?.append(preTitleSpacerView.widthAnchor.constraint(equalToConstant: 0))
```

Comment:
It'd also probably be better to do `preTitleSpacerView.isHidden = usesLeadingTitle` instead. Setting the width to 0 sounds like a bug waiting to happen.